### PR TITLE
[ED] Disclosed contract validation in Sandbox-on-X [DPP-1099]

### PR DESCRIPTION
**Pull request by tudor-da**
changelog_begin
changelog_end

### CODE REVIEW

File: ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/bridge/validate/ConflictCheckWithCommittedSpec.scala [hunk 1]
```diff
@@ -98,52 +100,135 @@ class ConflictCheckWithCommittedSpec
...
+      case failure => fail(s"Expectation mismatch: got $failure")
+    }
+  }
 
-    conflictCheckWithCommitted(input)
-      .map {
-        case Left(DuplicateKey(actualDuplicateKey)) => actualDuplicateKey shouldBe keyCreated
-        case failure => fail(s"Expectation mismatch: got $failure")
-      }
+  it should "fail validation mismatching let in disclosed contract" in new TestContext {
```
Comment (simonmaxen-da) [inline ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/bridge/validate/ConflictCheckWithCommittedSpec.scala:153]: Do you have a test case where a disclosed contract is valid ?

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/ContractStore.scala [hunk 2]
```diff
@@ -39,6 +39,12 @@ trait ContractStore {
   def lookupMaximumLedgerTimeAfterInterpretation(ids: Set[ContractId])(implicit
       loggingContext: LoggingContext
   ): Future[MaximumLedgerTime]
+
+  def lookupContractAfterInterpretation(
```
Comment (simonmaxen-da) [inline ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/ContractStore.scala:43]: Not sure about this name.  Could be verbosely called `lookupContractWithCreatedAt` which is what it technically does.  As it is not intended to be general purpose for for validation maybe `lookupContractForValidation` ?

**Review by simonmaxen-da — APPROVED** [review]
A couple of minor/optional comments.

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/ContractStore.scala (hunk 2 above)
Comment (tudor-da) [reply ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/ContractStore.scala:43]: `lookupContractForValidation` sounds better. Thanks!

**Pull request merged**